### PR TITLE
Fixes issue #177

### PR DESCRIPTION
**Pull request by kongdigital**
Also adds support hex decimals
Fixes issue #177

### CODE REVIEW

**Comment by kongdigital** [comment]
@jquense Any feedback?

**Comment by bluedaniel** [comment]
Is this library still being maintained?

**Comment by jquense** [comment]
Thanks for jumping there @bluedaniel with that very helpful comment

**Comment by bluedaniel** [comment]
Sorry @jquense but I thought it was a valid question seeing as though the PR is almost 2 months old.

File: src/number.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import isAbsent from './util/isAbsent';
 let isNaN = value => value != +value;
 
 let isInteger = val => isAbsent(val) || val === (val | 0);
+// let isInfinite = val => Math.abs(val) === Number.POSITIVE_INFINITY;
```
Comment (jquense) [inline src/number.js]: can we remove the commented code?

File: src/number.js [hunk 2]
```diff
@@ -17,7 +18,10 @@ export default function NumberSchema() {
       if (this.isType(value)) return value;
 
       let parsed = parseFloat(value);
-      if (this.isType(parsed)) return parsed;
+      if (this.isType(parsed) && isFinite(value)) return parsed;
+
+      if (typeof value === 'string' && value !== '' && this.isType(+value))
+        return +value;
```
Comment (jquense) [inline src/number.js]: Sorry for the delay, I've been trying to think what the best option here is in terms of fixing number parsing more generally.

Rereading everything, it seems like `Number(value)` is the best bet (as you have here) and we remove the parseFloat case. What do you think?

File: src/number.js (hunk 1 above)
Comment (kongdigital) [reply src/number.js]: 👍

File: src/number.js (hunk 2 above)
Comment (kongdigital) [reply src/number.js]: ActuallyI think it has a problem now that I've revisited it. I realised there is no test for whitespace characters.

**Comment by MaxMilton** [comment]
Hey i'd love to see this feature, is there anything I can do to help get it merged?

**Comment by jquense** [comment]
thanks!

**Comment by MaxMilton** [comment]
❤️ Thanks for the merge! It was holding us back on a big project.

**Pull request merged**